### PR TITLE
Azure Monitor Logs bug fixes

### PR DESCRIPTION
**Pull request by JustinMDotNet**
Added filter logic to exclude comments in MonitorClient. Added error handling in KustoClient to handle ErrorResponseExceptions.

### CODE REVIEW

File: src/Microsoft.Kusto.ServiceLayer/DataSource/Monitor/MonitorClient.cs [hunk 1]
```diff
@@ -61,12 +64,36 @@ private HttpClient GetHttpClient(string token)
...
+            {
+                var message = $"{ex.Body.Error.Innererror.Message} {ex.Body.Error.Innererror?.Innererror?.Message}";
+                throw new Exception(message, ex);
+            }
+        }
+
+        private string MinimizeQuery(string query)
+        {
+            var script = CodeScript.From(query, GlobalState.Default);
+            return script.Blocks[0].Service.GetMinimalText(MinimalTextKind.RemoveLeadingWhitespaceAndComments);
```
Comment (JustinMDotNet) [inline src/Microsoft.Kusto.ServiceLayer/DataSource/Monitor/MonitorClient.cs:96]: Remove leading whitespaces and comments from queries.

File: src/Microsoft.Kusto.ServiceLayer/DataSource/Monitor/MonitorClient.cs [hunk 2]
```diff
@@ -61,12 +64,36 @@ private HttpClient GetHttpClient(string token)
...
-            return await _queryClient.QueryAsync(query, cancellationToken: cancellationToken);
+            try
+            {
+                var minimizedQuery = MinimizeQuery(query);
+                return await _queryClient.QueryAsync(minimizedQuery, cancellationToken: cancellationToken);
+            }
+            catch (ErrorResponseException ex)
+            {
+                var message = $"{ex.Body.Error.Innererror.Message} {ex.Body.Error.Innererror?.Innererror?.Message}";
+                throw new Exception(message, ex);
```
Comment (JustinMDotNet) [inline src/Microsoft.Kusto.ServiceLayer/DataSource/Monitor/MonitorClient.cs:75]: Catch ErrorResponseException and parse error message to display to user in ADS.

File: src/Microsoft.Kusto.ServiceLayer/DataSource/Monitor/MonitorClient.cs [hunk 3]
```diff
@@ -61,12 +64,36 @@ private HttpClient GetHttpClient(string token)
...
-            return _queryClient.Query(query);
+            try
+            {
+                var minimizedQuery = MinimizeQuery(query);
+                return _queryClient.Query(minimizedQuery);
+            }
+            catch (ErrorResponseException ex)
+            {
+                var message = $"{ex.Body.Error.Innererror.Message} {ex.Body.Error.Innererror?.Innererror?.Message}";
+                throw new Exception(message, ex);
```
Comment (JustinMDotNet) [inline src/Microsoft.Kusto.ServiceLayer/DataSource/Monitor/MonitorClient.cs:89]: Catch ErrorResponseException and parse error message to display to user in ADS.

File: src/Microsoft.Kusto.ServiceLayer/DataSource/Monitor/MonitorDataSource.cs [hunk 4]
```diff
@@ -94,7 +94,7 @@ private void SetupColumns(TablesModel table, DataSourceObjectMetadata tableNodeI
 
         public override async Task<IDataReader> ExecuteQueryAsync(string query, CancellationToken cancellationToken, string databaseName = null)
         {
-            var results =  await _monitorClient.QueryAsync(query, cancellationToken);
+            var results = await _monitorClient.QueryAsync(query, cancellationToken);
```
Comment (JustinMDotNet) [inline src/Microsoft.Kusto.ServiceLayer/DataSource/Monitor/MonitorDataSource.cs:97]: Removed empty space.

**Review by qifahs — APPROVED** [review]

**Pull request merged**